spark_log_parser/cli.py

```diff
@@ -43,10 +43,17 @@ def main():
     with tempfile.TemporaryDirectory() as work_dir:
 
         event_log_paths = Extractor(
-            unquote(args.log_file.resolve().as_uri()), work_dir, thresholds=ExtractThresholds(size=20000000000)
+            unquote(args.log_file.resolve().as_uri()),
+            work_dir,
+            thresholds=ExtractThresholds(size=20000000000),
         ).extract()
 
-        event_log = EventLogBuilder(event_log_paths, work_dir).build()
+        event_log, parsed = EventLogBuilder(event_log_paths, work_dir).build()
+
+        if parsed:
+            print("Input log is already parsed")
+            return
```

Comment:
I think this constitutes an abnormal exit since we're not putting the result in the result directory. In which case we can simply do,
```
return "Input already contains a parsed log file"
```
The invocation is wrapped in a [sys.exit](https://docs.python.org/2/library/sys.html#sys.exit) in the script that's created when the package is installed:
```
% cat ~/.virtualenvs/sync_backend/bin/spark-log-parser
#!/Users/scottromney/.virtualenvs/sync_backend/bin/python
# -*- coding: utf-8 -*-
import re
import sys
from spark_log_parser.cli import main
if __name__ == '__main__':
    sys.argv[0] = re.sub(r'(-script\.pyw|\.exe)?$', '', sys.argv[0])
    sys.exit(main())
```
So returning the string will result in a message printed to stderr and an exit code of 1 as expected.

Alternatively, we could treat this as a normal scenario and copy the parsed event log to the result directory (it might be buried in an archive after all) logging a warning to indicate that it was already parsed. That way this function would continue to have a single point of return, but I think this behavior is more in keeping with this command line utility as a parser only - when it can't parse you're using it incorrectly and should expect an error.

Reply:
Nice good call, I changed it to copy over the parsed logs to the destination regardless.